src/caffeine/prif_queries_s.f90

```diff
@@ -13,6 +16,15 @@
   end procedure
 
   module procedure prif_base_pointer
+    integer(c_int) :: num_img
+
+    call prif_num_images(image_count=num_img)
+    call assert(image_num .ge. 0 .and. image_num .le. num_img, "prif_base_pointer: image_num not within valid range")
```

Comment:
Ok, so now the assertion appears to be logically correct.

Unfortunately as currently expressed the Fortran optimizer is unlikely be able to remove the associated overhead, because the call to `prif_num_images` currently includes an unconditional call to `caf_num_images() bind(C)` that is cross-module and cross-language, and this then invokes libgasnet (cross-library). This call chain is ridiculously over-complicated for the need at-hand (because the image data in the initial team is invariant for a given process over the lifetime of the job) and likely opaque to most analyses.

Ideas: 
1. `prif_init` should store such process info (image count and index in the initial team) as `prif` module variables in Fortran and just read them here, which should be alot more efficient and optimizable.
2. `caf_caffeinate` should also store this info as static variables in C-land so we also have very efficient access to that information in C.
3. We could optionally move or duplicate this assertion into `caf_convert_base_addr`, where we have hard guarantees that assertions compile away to nothing in production builds.

I'd really like to see options 1 and 2 deployed ASAP, as we'll eventually need similar checks elsewhere.

Reply:
As discussed in our meeting today, we will hold off on this change until after other critical changes. Issue #59 tracks this. Eventual implementation will change `num_img` in the above line of code to `initial_team%num_images`.